tidy3d/plugins/design/design.py

```diff
@@ -102,78 +144,146 @@ def run(self, function: Callable, **kwargs) -> Result:
             Can be converted to ``pandas.DataFrame`` with ``.to_dataframe()``.
         """
 
-        # run the function from the method
-        fn_args, fn_values = self.method.run(parameters=self.parameters, fn=function)
+        self._name_generator = self._create_name_generator()
+        next(self._name_generator)
 
-        fn_source = self.get_fn_source(function)
+        # Run based on how many functions the user provides
+        if fn_post is None:
+            fn_args, fn_values, aux_values = self.run_single(fn)
 
-        # package the result
-        return self._package_run_results(fn_args=fn_args, fn_values=fn_values, fn_source=fn_source)
+        else:
+            fn_args, fn_values, aux_values = self.run_pre_post(fn_pre=fn, fn_post=fn_post)
 
-    @staticmethod
-    def _make_batch_fn_source(fn_source_pre: str, fn_source_post: str) -> str:
-        """How to make the full function source from the pre and post functions."""
-
-        if (fn_source_pre is None) and (fn_source_post is None):
-            return None
+        fn_source = self.get_fn_source(fn)
 
-        return str(fn_source_pre) + "\n\n" + str(fn_source_post)
-
-    def run_batch(
```

Comment:
for backwards compatibility, do we want to just keep this but have it wrap the `run_pre_post` with a `log.warning()` about the new syntax?

Reply:
Changed. Discussion for later, are there a writing recommendations or a standardised "tone" for writing in tidy3D?

Reply:
Hm, not really, best is to take inspiration from the existing warnings and things. but for this, the tone I'd take would be something like.

```
In version 2.8.0, the 'run_batch' method is replaced by 'run_pre_post'. While the original syntax will still work, new functionality will be added to the new method, so we recommend update your scripts.
```

In general I think we want to avoid referencing ourselves, eg using "We".